aten/src/ATen/native/cudnn/Conv_v8.cpp

```diff
@@ -355,22 +355,98 @@ void generate_and_filter_plans(const cudnnHandle_t handle, cudnn_frontend::Opera
     for (auto &plan : valid_plans) {
       if (plan.getWorkspaceSize() <= max_workspace_size) {
         new_valid_plans.emplace_back(std::move(plan));
-        plan_count++;
-      }
-      if (max_plans && plan_count >= max_plans) {
-        break;
       }
     }
     valid_plans = std::move(new_valid_plans);
   }
 }
 
+// TODO: This is a hacked version of a cuDNN frontend function to allow us to
+// limit the number of benchmarked plans to save time. Once upstream cuDNN
+// frontend adds this functionality, remove this duplicate function definition.
+template <cudnn_frontend::CudnnFindSamplingTechnique samplingTechnique>
+auto
+temp_cudnn_time_sorted_plan_hack(cudnnHandle_t handle, cudnn_frontend::executionPlans_t plans, cudnn_frontend::VariantPack const &variantPack, const unsigned int benchmark_limit) -> cudnn_frontend::executionPlans_t {
+    cudnn_frontend::executionPlans_t time_sorted_plans;
+
+    auto plan_cmp = [](const cudnn_frontend::ExecutionPlan& a, const cudnn_frontend::ExecutionPlan& b) {return a.getExecutionTime() < b.getExecutionTime();};
+    std::set<std::reference_wrapper<cudnn_frontend::ExecutionPlan>, decltype(plan_cmp)> timed_execution_plans(plan_cmp);
+
+    const int maxIterCount =
+        (samplingTechnique == cudnn_frontend::CudnnFindSamplingTechnique::CUDNN_FIND_SAMPLE_ONCE)
+            ? 1
+            : (samplingTechnique == cudnn_frontend::CudnnFindSamplingTechnique::CUDNN_FIND_SAMPLE_MEDIAN_OF_THREE) ? 3 : 100;
+    const float threshhold = 0.95f;
+
+    cudaEvent_t start, stop;
+    cudaEventCreate(&start);
+    cudaEventCreate(&stop);
+    cudaDeviceSynchronize();
+
+    cudaStream_t stream = nullptr;
+    cudnnGetStream(handle, &stream);
+
+    unsigned int count = 0;
+
+    for (auto &plan : plans) {
+        float time_ms       = 0.0f;
+        float final_time_ms = 0.0f;
+        float min_time_ms   = std::numeric_limits<float>::max();
+
+        // Warm-up run
+        auto warmup_status = ::cudnnBackendExecute(handle, plan.get_raw_desc(), variantPack.get_raw_desc());
+        if (warmup_status != CUDNN_STATUS_SUCCESS) {
+            continue;
+        }
+
+        cudaDeviceSynchronize();
+
+        for (int i = 0; i < maxIterCount; i++) {
+            cudaEventRecord(start, stream);
+
+            cudnnBackendExecute(handle, plan.get_raw_desc(), variantPack.get_raw_desc());
+
+            cudaEventRecord(stop, stream);
+            cudaEventSynchronize(stop);
+            cudaEventElapsedTime(&time_ms, start, stop);
+
+            if (samplingTechnique == cudnn_frontend::CudnnFindSamplingTechnique::CUDNN_FIND_SAMPLE_TILL_STABLE) {
+                final_time_ms = std::min(min_time_ms, time_ms);
+                if (time_ms / min_time_ms < threshhold) {
+                    min_time_ms = final_time_ms;
+                } else {
+                    break;
+                }
+            } else {
+                final_time_ms = i == (maxIterCount / 2) ? time_ms : final_time_ms;
```

Comment:
nit: `maxIterCount >= 2` is more readable

Reply:
I'm not sure I understand which condition should be changed? Is it the part using the median time (3/2 = 1) , (1/2 = 0)?

This is taken from the cuDNN frontend: https://github.com/NVIDIA/cudnn-frontend/blob/81a041a68245cd8f871c43bbbbd5b6b627979a30/include/cudnn_frontend_find_plan.h#L94 which at first reading looks like it throws out the last iteration in "median of three" which seems unexpected.

Reply:
Ah ok so it's not really MEDIAN_OF_THREE, it's just time on the second iteration. Fine, leave it like this.